google-cloud-bigquery/lib/google/cloud/bigquery/table.rb

```diff
@@ -118,6 +118,16 @@ def table_ref
           table_ref
         end
 
+        ###
+        # The gapi fragment containing TimePartitioning ExpirationMS and Type
+        # as a camel-cased hash.
+        def time_partitioning
+          ensure_full_data!
+          time_part = @gapi.time_partitioning
+          time_part = time_part.to_hash if time_part.respond_to? :to_hash
+          time_part
```

Comment:
Can the return type be a structured Ruby object, instead of a Hash? This should also be the parameter type for `time_partitioning=`, below.